src/Type/Definition/FieldsTrait.php

```diff
@@ -94,15 +97,22 @@ protected function buildFieldMap($fieldsOrThunk): array
                     null === $fieldConfig['resolve'] || \is_callable($fieldConfig['resolve']),
                     \sprintf(
                         '%s.%s field resolver must be a function if provided, but got: %s',
-                        $this->getName(),
+                        $this->name,
                         $fieldName,
                         toString($fieldConfig['resolve'])
                     )
                 );
             }
 
-            $fieldConfig['name'] = $fieldName;
-            $fieldMap[$fieldName] = new Field($fieldConfig);
+            $fieldMap[$fieldName] = new Field(
+                $fieldName,
+                $fieldConfig['description'] ?? null,
```

Comment:
I think maybe we wnat to get rid of `ConfigAwareTrait` but this kind of workaround smells bad to me :-?